### PR TITLE
Feature | New Component SFBadge

### PR DESCRIPTION
**Pull request by Yopo-Nicolas**

### CODE REVIEW

File: src/Components/SFBadge/SFBadge.tsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+          </svg>
+        )
+      ) : undefined;
+    } else return undefined;
+  };
+
+  return (
+    <StyledBadge
+      {...props}
+      anchorOrigin={{ vertical: 'top', horizontal: 'left' }}
```
Comment (pdcastano) [inline src/Components/SFBadge/SFBadge.tsx:86]: This value should be set as the default behavior, not a fixed value.

File: src/Components/SFBadge/SFBadge.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+      className={`${size}`}
```
Comment (pdcastano) [inline src/Components/SFBadge/SFBadge.tsx:87]: this should have className and size as value.

File: src/Components/SFBadge/SFBadge.tsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+import {
+  SFCommonBlack,
+  SFCommonWhite,
+  SFRedMainDark,
+  SFRedMainLight
+} from '../..';
+
+const StyledBadge = withStyles((theme: Theme) => ({
+  root: {
+    '&.small': {
```
Comment (pdcastano) [inline src/Components/SFBadge/SFBadge.tsx:14]: image
Small badge are not well styled

**Review by pdcastano — CHANGES_REQUESTED** [review]

File: src/Components/SFBadge/SFBadge.tsx (hunk 1 above)
Comment (Yopo-Nicolas) [reply src/Components/SFBadge/SFBadge.tsx]: changed

File: src/Components/SFBadge/SFBadge.tsx (hunk 2 above)
Comment (Yopo-Nicolas) [reply src/Components/SFBadge/SFBadge.tsx]: changed

File: src/Components/SFBadge/SFBadge.tsx (hunk 3 above)
Comment (Yopo-Nicolas) [reply src/Components/SFBadge/SFBadge.tsx:14]: fixed layout

File: src/Components/SFBadge/SFBadge.tsx [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    },
+
+    '&.medium': {
+      '& .MuiBadge-badge': {
+        width: '21px',
+        height: '21px'
+      }
+    }
+  },
+  badge: {
```
Comment (pdcastano) [inline src/Components/SFBadge/SFBadge.tsx:29]: image
The badge is not displayed in the correct position.
Check https://mui.com/components/badges/#unstyled to see how it works.

File: src/Components/SFBadge/SFBadge.tsx [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+              fill='white'
+            />
+          </svg>
+        )
+      ) : undefined;
+    } else return undefined;
+  };
+
+  return (
+    <StyledBadge {...props} className={size} badgeContent={BadgeContent()}>
```
Comment (pdcastano) [inline src/Components/SFBadge/SFBadge.tsx:87]: You are missing the SFBadge className prop in this way. You are replacing it with the size value instead.

**Review by pdcastano — CHANGES_REQUESTED** [review]

File: src/Components/SFBadge/SFBadge.tsx (hunk 4 above)
Comment (Yopo-Nicolas) [reply src/Components/SFBadge/SFBadge.tsx:29]: fixed

File: src/Components/SFBadge/SFBadge.tsx (hunk 5 above)
Comment (Yopo-Nicolas) [reply src/Components/SFBadge/SFBadge.tsx]: added

File: src/Components/SFBadge/SFBadge.tsx [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+    lineHeight: '14px',
+    color: theme.palette.type === 'light' ? SFCommonWhite : SFCommonBlack,
+
+    '& .badgeDot': {
+      fill: theme.palette.type === 'light' ? SFCommonWhite : SFCommonBlack
+    }
+  },
+  anchorOriginTopLeftCircle: {
+    transform: 'scale(1) translate(-55%, -55%)'
+  }
```
Comment (pdcastano) [inline src/Components/SFBadge/SFBadge.tsx:44]: Remove this style. As you can see, this applies only for one anchorOrigin position and that is wrong.
On the other side, checking the stories, I think that we will have to adjust the badge position manually in each particular case into the application.

**Review by pdcastano — CHANGES_REQUESTED** [review]

**Comment by pdcastano** [comment]
LGTM

**Review by pdcastano — APPROVED** [review]

**Pull request merged**